### DISCUSSION TITLE
Slow Data Entry Performance Issue in Firefly III Deployment on Synology NAS

### DISCUSSION THREAD

**Discussion by xkmttc** [firefly-iii/firefly-iii#8710] [Firefly III]
Hello everyone,

I've deployed Firefly III on my Synology NAS, and I find it very useful and enjoyable to use. However, I've encountered an issue while using it. When I enter data, such as expenses, there's a significant delay of about 5 seconds before the data is finally written. This happens consistently whether I'm using the mobile app, the web interface, or even shortcut commands. It seems to be slower than expected. I'm wondering if this is normal behavior. Thank you all for your help.

**Comment by JC5** [comment]
No it's not normal, no. It should be less then a second.

Could you check out the logs and see what it hangs on?

https://docs.firefly-iii.org/how-to/general/debug/

**Reply by JC5** [reply]
Sorry, this doesn't tell me what I need to know. Please follow the instructions so I see the debug logs :)
